File: src/engine/commands/shoutcommand.cpp
#include "shoutcommand.h"


#define super Command

ShoutCommand::ShoutCommand(QObject *parent) :
    super(parent) {

    setDescription("Shout something. Everyone in the same room and every "
                   "adjacent room can hear it.\n"
                   "\n"
                   "Example: shout Anyone there?");
}

ShoutCommand::~ShoutCommand() {
}

void ShoutCommand::execute(Player *player, const QString &command) {

    super::prepareExecute(player, command);

    if (!assertWordsLeft("Shout what?")) {
        return;
    }

    QString message = takeRest();

    player->shout(message);
}

File: data/commands/shoutcommand.js
function ShoutCommand() {
    Command.call(this);

    this.setDescription("Shout something. You should shout when you want to be heard about " +
                        "greater distances.\n" +
                        "\n" +
                        "Example: shout Anyone there?");
}

ShoutCommand.prototype = new Command();
ShoutCommand.prototype.constructor = ShoutCommand;

ShoutCommand.prototype.execute = function(player, command) {

    this.prepareExecute(player, command);

    if (!this.assertWordsLeft("Shout what?")) {
        return;
    }

    var message = takeRest();

    player.shout(message);
}

CommandRegistry.registerCommand("shout", new ShoutCommand());
